### PR TITLE
feat(deploy-tooling): add auth type to cli params

### PR DESCRIPTION
**Pull request by cianmSAP**
Add `--authentication-type` to the available CLI params in deploy tooling (considerations for reentrance tickets)

### CODE REVIEW

File: packages/deploy-tooling/src/cli/index.ts [hunk 1]
```diff
@@ -32,6 +32,7 @@ export function createCommand(name: 'deploy' | 'undeploy'): Command {
         .addOption(new Option('--client <sap-client>', 'Client number of target ABAP system').conflicts('destination'))
         .addOption(new Option('--cloud', 'Target is an ABAP Cloud system').conflicts('destination'))
         .addOption(new Option('--service <service-path>', 'Target alias for deployment service'))
+        .addOption(new Option('--authentication-type <authentication-type>', 'Authentication type for the app'))
```
Comment (longieirl) [inline packages/deploy-tooling/src/cli/index.ts:35]: The readme.md was updated recently, you will need to adjustthe `options` to include this new param as well.

Comment (cianmSAP) [reply packages/deploy-tooling/src/cli/index.ts:35]: Updated

**Review by longieirl — DISMISSED** [review]
Change look good.
Test coverage is good.
`--authentication-type` is not conflicting with other params.

File: packages/deploy-tooling/test/unit/cli/index.test.ts [hunk 2]
```diff
@@ -79,6 +79,7 @@ describe('cli', () => {
         ];
 
         const cliCmdWithUaa = [...cliCmd, '--cloud-service-env', '--service', '/bc/my/uaa/deploy/service'];
+        const cliCmdWithAuthTyp = [...cliCmd, '--authentication-type', 'reentranceTicket'];
```
Comment (mikicvi-SAP) [inline packages/deploy-tooling/test/unit/cli/index.test.ts:82]: @cianmSAP  `cliCmdWithAuthTyp`, is this a typo?

**Review by mikicvi-SAP — CHANGES_REQUESTED** [review]

**Review by mikicvi-SAP — APPROVED** [review]
Changes look good
Thanks for correction of the typo @cianmSAP

**Review by devinea — APPROVED** [review]
code changes lg.
changeset ✅

**Review by longieirl — APPROVED** [review]
re-approving.

**Pull request merged**